datafusion-examples/examples/dynamic_csv_sql.rs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use datafusion::error::Result;
+use datafusion::prelude::*;
+
+/// This example demonstrates executing a simple query against an Arrow data source (CSV) and
+/// fetching results
+#[tokio::main]
+async fn main() -> Result<()> {
+    // create local execution context
+    let ctx = SessionContext::new();
+
+    let testdata = datafusion::test_util::arrow_test_data();
+    let path = &format!("file:///{testdata}/csv/aggregate_test_100.csv");
+    // execute the query
+    let df = ctx
+        .sql(
+            format!(
+                r#"SELECT column_1, MIN(column_12), MAX(column_12)
```

Comment:
The column is `c1` actually. There're some issues about getting CSV header automatically.

Reply:
ok, I found that this is the default behavior for `ListTable`. The dynamic query in datafusion-cli is the same as this. I think we don't need to change it in this PR.

Reply:
This behavior is controlled by the config option in the current session context. We can create the session context like
```rust
let cfg = SessionConfig::new().set_str("datafusion.catalog.has_header", "true");
let ctx = SessionContext::new_with_config(cfg);
```
to enable the header scanning.